addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from collections import defaultdict
+
+from odoo import _, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class MailTrackingDurationMixin(models.AbstractModel):
+    _name = "mail.tracking.duration.mixin"
+    _description = "Mixin to compute the time a record spent in each stage"
+    _track_duration_field = "stage_id"
```

Comment:
Not sure we should give a default value (I prefer people to add it explicitly), but that's personal taste :)

Reply:
Ok, I added it in the compute docstring instead